packages/admin-ui/src/Checkbox/useCheckbox.ts

```diff
@@ -0,0 +1,46 @@
+import { useEffect, useMemo, useState } from "react";
+import { autorun } from "mobx";
+import { CheckboxProps } from "./Checkbox";
+import { CheckboxPresenter, CheckboxPresenterParams } from "./CheckboxPresenter";
+
+export const useCheckbox = (props: CheckboxProps) => {
+    const params: CheckboxPresenterParams = useMemo(
+        () => ({
+            id: props.id,
+            label: props.label,
+            value: props.value,
+            checked: props.checked,
+            indeterminate: props.indeterminate,
+            disabled: props.disabled,
+            onCheckedChange: props.onCheckedChange
+        }),
+        [
+            props.id,
+            props.label,
+            props.value,
+            props.checked,
+            props.indeterminate,
+            props.disabled,
+            props.onCheckedChange
+        ]
+    );
+    const presenter = useMemo(() => {
+        const presenter = new CheckboxPresenter();
+        presenter.init(params);
+        return presenter;
+    }, []);
+
+    const [vm, setVm] = useState(presenter.vm);
+
+    useEffect(() => {
+        presenter.init(props);
+    }, [params, presenter]);
```

Comment:
Is `presenter` needed here? from the looks of it, it never changes because cache key in L31?

```suggestion
    }, [params]);
```

Reply:
👍🙏